### ISSUE TITLE
`selector-no-type` doesn't work with nested CSS when & is used as a prefix

### ISSUE THREAD

**Issue by onigoetz** [stylelint/stylelint#834]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When I have nested rules, selector-no-type doesn't work correctly

> Which rule, if any, is this issue related to?

`selector-no-type`

> What CSS is needed to reproduce this issue?

```css
.MyOtherComponent {
    &__element {
        color: orange;
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-type": true
  }
}
```

> Which version of stylelint are you using?

 `4.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Yes, CLI with the config shown before

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to nesting

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
test.scss
2:5    Unexpected type selector (selector-no-type)
```

**Comment by jeddy3** [comment]
@onigoetz Thanks for the detailed report :)

This seems to be specific to when the nesting selector is used as a prefix. It's a quick fix and there's a pending PR for it.

**Comment by onigoetz** [comment]
That was fast ! thanks guys
